packages/legend-studio/src/stores/editor-state/element-editor-state/mapping/MappingEditorState.ts

```diff
@@ -936,6 +948,21 @@ export class MappingEditorState extends ElementEditorState {
       ) {
         flatData_setSourceRootRecordType(setImplementation, newSource);
       } else {
+        if (setImplementation instanceof InstanceSetImplementation) {
```

Comment:
not only we need to do this, but we also need to modify the `if` blocks above this, so the top level conditionals only check for the tpytype of the set implementation, for example

```ts
if (setImplementation instanceof PureInstanceSetImplementation) {
  if (newSource instanceof Class || newSource === undefined) {
    pureInstanceSetImpl_setSrcClass(setImplementation, newSource);
  }
} else if (
...
```